src/nodes/node.h

```diff
@@ -45,6 +45,58 @@ class Node : public Serialisable<>
     // Return short summary of the node's purpose
     virtual std::string_view summary() const = 0;
 
+    /*
+     * Node message
+     */
+
+    // Enumeration for message status
+    enum class MessageStatus
+    {
+        Any,
+        Info,
+        Warn,
+        Error,
+        Exception
```

Comment:
We talked about not needing to store exceptions, so:
```suggestion
        Error
```